File: DEC19B/BINADD.py
t = int(input())


def binaryToDecimal(n):
    n = n[-1::-1]
    decimal = 0
    for i in range(len(n)):
        decimal += int(n[i]) * (2**i)
    return decimal


for _ in range(t):
    a = binaryToDecimal(input())
    b = binaryToDecimal(input())
    counter = 0
    while b>0:
        U = a ^ b
        V = a & b
        a = U
        b = V*2
        counter += 1
    print(counter)

File: DEC19B/BINADD.cpp
#include<bits/stdc++.h>

using namespace std;

long long int binary_to_decimal(string n)
{
    long long int decimal = 0;
    for(int i = n.length()-1, j=0; i>=0; i--, j++)
    {
        decimal += (int(n[j]) - 48) * pow(2, i);
    }
    return decimal;
}

int main()
{
    int t;
    cin >> t;
    cin.ignore();
    while(t)
    {
        long long int da, db, U, V;
        string a, b;
        getline(cin, a);
        getline(cin, b);
        da = binary_to_decimal(a);
        db = binary_to_decimal(b);
        // cout << da << '\t' << db << endl;
        long long int counter = 0;
        while(db>0)
        {
            ++counter;
            U = da ^ db;
            V = da & db;
            da = U;
            db = V * 2;
            // cout << U << "\t" << V << endl;
        }
        cout << counter << endl;
        t--;
    }
    return 0;
}